transcom/mymove
# [MB-16924] Update `lines_of_accounting.loa_trnsn_id` values

## [MB-16924](https://dp3.atlassian.net/browse/MB-16924)

## Summary

Adds a python script that can be used on the spreadsheet in keeper to generate the secure migration script locally.

Since the file is large, expect it to takes a minute or two to read the file.

There's no need to update the payment request generation, as it is already mapped to the correct column [here](https://github.com/transcom/mymove/blob/4fe7f085d7877190e56b737e1cd4ebdc93ed0cf4/pkg/services/invoice/ghc_payment_request_invoice_generator.go#L764)

- [x] [Upload migration to S3](https://github.com/transcom/mymove/pull/11271)


## Verification Steps for the Author

These are to be checked by the author.

- [x] Tested locally with deployed_migrations

## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Setup to Run the Code

### How to test

Test generation of secure migration (requires keeper access):

1. Download the file from keeper (see Jira ticket)
2. Setup python locally, however you prefer to do so e.g.:
   * setup venv inside project directory
   * `source ./venv/bin/activate`
   * `pip install pandas`
   * `pip install openpyxl`
   * `python3 ./cmd/load-spreadsheet-data/load-updated-loa-trnsn-ids.py "./tmp/Line Of Accounting(U).xlsx"`

Test application of dummy migration:

1. `make db_dev_fresh` should take care of it.
2. Confirm that your `lines_of_accounting` table has `loa_trnsn_id` values after migrations successfully run. You should see the following data:
   <img width="150" alt="image" src="https://github.com/transcom/mymove/assets/15805554/7e913ce7-567c-42b7-93e5-8254f9c4bb78">

   Additionally, the loa_trnsn_id column should now be varchar(3):
   <img width="240" alt="image" src="https://github.com/transcom/mymove/assets/15805554/ea432166-396f-4594-a4ce-4f838f1a390c">


Test the secure migration itself with the other prod migrations:
* https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#secure-migrations

### Database

#### Any new migrations/schema changes:


[MB-16924]: https://dp3.atlassian.net/browse/MB-16924?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
